compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -602,7 +602,9 @@ public boolean isSubTypeOfBaseType(BType type, int baseTypeTag) {
         if (TypeTags.isXMLTypeTag(baseTypeTag)) {
             return true;
         }
-        return ((BUnionType) type).getMemberTypes().stream().allMatch(memType -> memType.tag == baseTypeTag);
+        return ((BUnionType) type).getMemberTypes().stream().allMatch(memType -> memType.tag == baseTypeTag ||
```

Comment:
Is it correct to change this without changing L599?

Reply:
Fixed in 5a9b935